File: app/src/main/java/com/snaptiongame/app/presentation/view/listeners/ItemListener.java
package com.snaptiongame.app.presentation.view.listeners;

import android.support.v7.widget.RecyclerView;

/**
 * @author Brian Gouldsberry
 */

public interface ItemListener {
    void updateUpvote(boolean value, int index);

    void updateFlag(int index, RecyclerView.ViewHolder holder);
}

File: app/src/main/java/com/snaptiongame/app/presentation/view/listeners/ItemListener.kt
package com.snaptiongame.app.presentation.view.listeners

import android.support.v7.widget.RecyclerView

/**
 * @author Brian Gouldsberry
 */

interface ItemListener {
    fun updateUpvote(value: Boolean, index: Int)

    fun updateFlag(index: Int, holder: RecyclerView.ViewHolder)
}
